lib/Runtime/Types/PathTypeHandler.cpp

```diff
@@ -1418,18 +1421,10 @@ namespace Js
                     swprintf_s(reason, 1024, _u("OffsetOfInlineSlot mismatch. Required = %d, Cached = %d"), this->GetOffsetOfInlineSlots(), cachedDynamicTypeHandler->GetOffsetOfInlineSlots());
 #endif
                 }
-                else if (cachedDynamicTypeHandler->GetInlineSlotCapacity() !=
-                         (useObjectHeaderInlining ?
-                         DynamicTypeHandler::RoundUpObjectHeaderInlinedInlineSlotCapacity(requestedInlineSlotCapacity) :
-                         DynamicTypeHandler::RoundUpInlineSlotCapacity(requestedInlineSlotCapacity)))
+                else if (cachedDynamicTypeHandler->GetInlineSlotCapacity() != roundedInlineSlotCapacity)
                 {
-                    cachedDynamicType = nullptr;
-#if DBG
-                    uint16 requiredCapacity = useObjectHeaderInlining
-                        ? DynamicTypeHandler::RoundUpObjectHeaderInlinedInlineSlotCapacity(requestedInlineSlotCapacity)
-                        : DynamicTypeHandler::RoundUpInlineSlotCapacity(requestedInlineSlotCapacity);
-                    swprintf_s(reason, 1024, _u("InlineSlotCapacity mismatch. Required = %d, Cached = %d"), requiredCapacity, cachedDynamicTypeHandler->GetInlineSlotCapacity());
-#endif
+                    Assert(cachedDynamicTypeHandler->GetInlineSlotCapacity() >= roundedInlineSlotCapacity);
```

Comment:
Can you also confirm that it's safe to shrink the capacity in this way by verifying that the total number of properties <= the current inlineSlotCapacity?